### PR TITLE
Add Windows support for the Python API

### LINKED ISSUE #1360: [tracker] Windows-native TileDB-SOMA
**Describe the bug**
```
pip install tiledbsoma
Building wheels for collected packages: tiledbsoma
  Building wheel for tiledbsoma (pyproject.toml): started
  Building wheel for tiledbsoma (pyproject.toml): finished with status 'error'
Failed to build tiledbsoma
Note: you may need to restart the kernel to use updated packages.
  error: subprocess-exited-with-error
  
  Building wheel for tiledbsoma (pyproject.toml) did not run successfully.
  exit code: 1
  
  [5 lines of output]
  running bdist_wheel
  Checking: C:\Users\dawei bushen\AppData\Local\Temp\pip-install-725ip_cc\tiledbsoma_ce59cb0207fe481d9e0f336d8115e3fb\dist_links\dist\lib\tiledbsoma.dll exists: False
  Checking: C:\Users\dawei bushen\AppData\Local\Temp\pip-install-725ip_cc\tiledbsoma_ce59cb0207fe481d9e0f336d8115e3fb\dist_links\dist\bin\tiledbsoma.dll exists: False
  Could not find module 'tiledbsoma.dll' (or one of its dependencies). Try using the full path with constructor syntax.
  error: [WinError 2] 系统找不到指定的文件。
  [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for tiledbsoma
ERROR: Could not build wheels for tiledbsoma, which is required to install pyproject.toml-based projects
```

**To Reproduce**
Provide a code example and any sample input data (e.g. an H5AD) as an attachment to reproduce this behavior.

**Versions (please complete the following information):**
 Python 3.10.9 | packaged by Anaconda, Inc. | (main, Mar  1 2023, 18:18:15) [MSC v.1916 64 bit (AMD64)]
Windows-10-10.0.19045-SP0

**Additional context**

### PR DESCRIPTION
**Pull request by teo-tsirpanis**
**Issue and/or context:**

**Changes:**

**Notes for Reviewer:**

[SC-35700](https://app.shortcut.com/tiledb-inc/story/35700/implement-windows-support-for-soma)

Fixes #1360.

### CODE REVIEW

File: libtiledbsoma/src/tiledbsoma/tiledbsoma [hunk 1]
```diff
@@ -33,15 +33,7 @@
...
-// TODO Uncomment after finishing Python and R bindings
-// #include "soma_collection.h"
-// #include "soma_dataframe.h"
-// #include "soma_dense_ndarray.h"
-// #include "soma_experiment.h"
-// #include "soma_measurement.h"
-// #include "soma_object.h"
-// #include "soma_sparse_ndarray.h"
-
+#include "tiledbsoma_export.h"
```
Comment (eddelbuettel) [inline libtiledbsoma/src/tiledbsoma/tiledbsoma:36]: I presume that is a cmake generated file?  Should we say so?  A bit weird to include something not in the sources.

Comment (teo-tsirpanis) [reply libtiledbsoma/src/tiledbsoma/tiledbsoma:36]: I will add a comment. We need this file to support exporting symbols in a cross-platform fashion (replacing [this](https://github.com/single-cell-data/TileDB-SOMA/blob/fe829ad4a9669f48c12a5198a96cea699a174231/libtiledbsoma/src/soma/logger_public.h#L44)). The Core already has one.

Comment (eddelbuettel) [reply libtiledbsoma/src/tiledbsoma/tiledbsoma:36]: Understood from what's in it but I think I have seen many other project have such `#define` statements in the normal include directories, no?

Comment (eddelbuettel) [reply libtiledbsoma/src/tiledbsoma/tiledbsoma:36]: Thanks for dealing with the Windows build by the way.  Much appreciated.

Comment (teo-tsirpanis) [reply libtiledbsoma/src/tiledbsoma/tiledbsoma:38]: > I have seen many other project have such `#define` statements in the normal include directories

We could do that, but having CMake generate the file automatically handles the varying platforms, compilers and static or dynamic linkage.

**Comment by teo-tsirpanis** [comment]
The only Windows test failure was fixed by TileDB-Inc/TileDB-Py#1863. This is ready for review.

**Comment by teo-tsirpanis** [comment]
CI on Windows is green. Failures seem to be network-related; one address it showed in the error message gives 404 to me as well.

**Comment by eddelbuettel** [comment]
There was an entirely unrelated hickup in the CI we use for R, I just took care of that and am keeping fingers crossed for the re-run that is now happening.

File: apis/python/setup.py [hunk 2]
```diff
@@ -151,7 +147,10 @@ def find_or_build_package_data(setuptools_cmd):
         # cause that cache to fall out of sync.
         #
         # See `.github/workflows/python-ci-single.yml` for configuration.
-        subprocess.run(["./bld"], cwd=scripts_dir)
+        if os.name == "nt":
+            subprocess.run(["pwsh.exe", "./bld.ps1"], cwd=scripts_dir, check=True)
```
Comment (nguyenv) [inline apis/python/setup.py:151]: @beroy is currently working on removing the `bld` script and porting the cmake commands over into `setup.py`. I don't think anything needs to be modified in this particular PR but just letting relevant parties be aware.

**Review by nguyenv — APPROVED** [review]
Thanks!

**Comment by johnkerl** [comment]
Merging after discussion with @teo-tsirpanis and @nguyenv . cc @beroy re the additional `pwsh.exe` script along with `bld` for his `setup.py` refactoring.

**Pull request merged**